### DISCUSSION TITLE
torch

### DISCUSSION THREAD

**Discussion by TheCodeCache** [flet-dev/flet#3245] [Packages] [answered]
https://pypi.org/project/torch/

**Comment by ndonkoHenri** [comment, marked as answer]
PyTorch can't be packaged for Flet mobile apps — this is a fundamental limitation of torch itself, not a Flet-specific gap:

- No source distribution on PyPI: torch publishes only pre-built platform wheels; there's no sdist for Flet's mobile build to compile from.
- Size: The libtorch/ATen native runtime is hundreds of MB even CPU-only (manylinux wheels are 426–532 MB; the smallest wheel is ~88 MB) — too large to ship in an app?
- Official mobile direction is different: PyTorch Mobile is deprecated; the supported on-device path is [ExecuTorch](https://pytorch.org/executorch/), which runs exported .pte models through a separate C++ runtime — it does not give you import torch on the device.

Possible alternative: run inference on a backend service and call it from the app.
